### PR TITLE
chore(ci): enforce 92% coverage gate

### LINKED ISSUE #193: Raise backend coverage to the 92% gate
Copilot instructions and the publish skill both document a 92% coverage threshold, but \`npm run test:coverage\` currently reports **83.11% lines / 88.52% branches** on \`main\`.

## Biggest gaps (from latest c8 report)

- \`src/server/routes.js\` — **32.73% lines**, 28.57% functions. This is the largest offender by far.
- \`src/server/index.js\` — 91.07% lines (close, a handful of uncovered branches).
- \`src/utils/agent-sessions.js\` — 85.06% lines.
- \`src/utils/git.js\` — 89.97% lines.

## Proposal

1. Add route-level tests for the uncovered paths in \`routes.js\` (especially agent endpoints and preview proxy edge cases).
2. Add a c8 threshold to \`package.json\` so the gate is enforced by \`test:coverage\`:

\`\`\`json
"c8": {
  "check-coverage": true,
  "lines": 92,
  "functions": 92,
  "branches": 85,
  "statements": 92
}
\`\`\`

3. Wire CI to fail if the gate slips.

## Context

Flagged while shipping the Review Mode + SessionsHub polish PR — a frontend-only PR couldn't move the needle, and the gate isn't currently enforced in code.

### PR DESCRIPTION
**Pull request by dorlugasigal**
Closes #193.

Enforces a 92% line coverage gate via c8 `check-coverage`, removes the redundant CI bash threshold check, and consolidates the c8 excludes into the single `c8` config block in package.json.

Also adds a new docs section on keeping the host machine awake/reachable while running TermBeam as a background service — covers Amphetamine + caffeinate (macOS), PowerToys Awake + adapter power management (Windows), and systemd-inhibit (Linux). README links to the new section.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,8 +9,8 @@
...
+    "test": "node -e \"const{execFileSync:r}=require('child_process'),{readdirSync:d,statSync:s}=require('fs'),{join:j}=require('path');function f(p){let a=[];for(const e of d(p)){const c=j(p,e);s(c).isDirectory()?a.push(...f(c)):e.endsWith('.test.js')&&!e.startsWith('e2e-')&&e!=='devtunnel-install.test.js'&&a.push(c)}return a}r(process.execPath,['--test','--test-timeout=180000',...f('test')],{stdio:'inherit'})\"",
+    "test:coverage": "c8 --exclude=src/tunnel/ --exclude=test --reporter=text --reporter=lcov --reporter=json-summary --reporter=json node -e \"const{execFileSync:r}=require('child_process'),{readdirSync:d,statSync:s}=require('fs'),{join:j}=require('path');function f(p){let a=[];for(const e of d(p)){const c=j(p,e);s(c).isDirectory()?a.push(...f(c)):e.endsWith('.test.js')&&!e.startsWith('e2e-')&&e!=='devtunnel-install.test.js'&&a.push(c)}return a}r(process.execPath,['--test','--test-timeout=180000','--test-reporter=spec','--test-reporter-destination=stdout',...f('test')],{stdio:'inherit'})\"",
```
[the remark this answers is not in the document]
Comment (dorlugasigal) [reply package.json]: Done in 327563c1 — removed the duplicated CLI flags; c8 config in package.json is now the single source of truth.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -53,8 +53,8 @@ jobs:
         run: |
           COVERAGE=$(node -e "const c = require('./coverage/coverage-summary.json'); console.log(c.total.lines.pct)")
           echo "Coverage: ${COVERAGE}%"
-          if (( $(echo "${COVERAGE} < 80" | bc -l) )); then
-            echo "::error::Coverage ${COVERAGE}% is below 80% threshold"
+          if (( $(echo "${COVERAGE} < 92" | bc -l) )); then
+            echo "::error::Coverage ${COVERAGE}% is below 92% threshold"
             exit 1
```
[the remark this answers is not in the document]
Comment (dorlugasigal) [reply .github/workflows/ci.yml]: Done in 327563c1 — dropped the entire 'Check coverage threshold' step. `c8.check-coverage: true` (with `lines: 92`) makes `npm run test:coverage` exit non-zero on its own.

File: packages/site/src/content/docs/running-in-background.md [hunk 3]
```diff
@@ -271,11 +348,12 @@ TermBeam requires Node.js 20 or higher. Verify with `node --version` before sett
 
 <!-- prettier-ignore -->
 :::tip[Which method should I use?]
+
 - **Quick test?** → `nohup`
 - **Dev machine?** → PM2 (easiest setup, great logs)
 - **Server/always-on?** → systemd or launchd (OS-native, starts on boot)
 - **Windows?** → Task Scheduler or NSSM
-:::
+  :::
```
[the remark this answers is not in the document]
Comment (dorlugasigal) [reply packages/site/src/content/docs/running-in-background.md]: Done in 327563c1 — added a blank line before the closing `:::` so prettier no longer treats it as list continuation. Starlight now recognizes the closing marker.

File: README.md [hunk 4]
```diff
@@ -137,6 +137,8 @@ The installer checks for [PM2](https://pm2.keymetrics.io/) (and offers to instal
 
 For systemd, launchd, and Windows Task Scheduler setup, see the [Running in Background docs](https://dorlugasigal.github.io/TermBeam/running-in-background/).
 
+> 💡 **Keep the host awake** so the service stays reachable while you're away. macOS: pair with [Amphetamine](https://apps.apple.com/app/amphetamine/id937984704) (process trigger on `node`) or wrap with `caffeinate -dims`. Windows: enable [PowerToys Awake](https://learn.microsoft.com/windows/powertoys/) and disable network adapter power saving. Linux: use `systemd-inhibit` in your unit file. See [Keeping the Host Awake](https://dorlugasigal.github.io/TermBeam/running-in-background/#keeping-the-host-awake-) for the full setup.
```
[the remark this answers is not in the document]
Comment (dorlugasigal) [reply README.md:140]: Done — updated the PR description to mention the new awake-host docs alongside the coverage gating.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -49,14 +49,6 @@ jobs:
...
-        run: |
-          COVERAGE=$(node -e "const c = require('./coverage/coverage-summary.json'); console.log(c.total.lines.pct)")
-          echo "Coverage: ${COVERAGE}%"
-          if (( $(echo "${COVERAGE} < 80" | bc -l) )); then
-            echo "::error::Coverage ${COVERAGE}% is below 80% threshold"
-            exit 1
-          fi
       - name: Coverage Report
         if: github.event_name == 'pull_request'
         uses: davelosert/vitest-coverage-report-action@3c50566c523e04813df28de8f7c48dd97d663f1c # v2
```
[the remark this answers is not in the document]
Comment (dorlugasigal) [reply .github/workflows/ci.yml:54]: The PR description has been updated to reflect the current approach — it now states that the redundant CI bash threshold check is removed and that `c8.check-coverage` (with `lines: 92`) enforces the gate via `npm run test:coverage`.
